File: frontend/src/data/pokedex/pokedex.transfer.ts
export interface PokedexTO {
    [key: string]: PokemonSpeciesTO
}

export interface PokemonSpeciesTO {
    name: string
    sprite: string
    types: PokemonTypesTO
    abilities: PokemonAbilitiesTO
    baseSpecies: number
    evolutions: number[]
}

export interface PokemonTypesTO {
    primary: string
    secondary: string | null
}

export interface PokemonAbilitiesTO {
    1: string
    2: string | null
    3: string | null
}

export type NaturesTO = string[]

File: backend/src/main/kotlin/net/namibsun/nuztrack/transfer/PokedexTransfer.kt
package net.namibsun.nuztrack.transfer

import net.namibsun.nuztrack.constants.enums.Games
import net.namibsun.nuztrack.util.getSpriteForGameAndPokemon

typealias PokedexTO = Map<Int, PokemonSpeciesTO>

data class PokemonSpeciesTO(
        val pokedexNumber: Int,
        val name: String,
        val sprite: String,
        val types: PokemonTypesTO,
        val abilities: Map<Int, String?>,
        val baseSpecies: Int,
        val evolutions: List<Int>,
        val genderRate: Int,
        val isLegendary: Boolean,
        val generation: Int
) {
    fun withGameSpecificSprite(game: Games): PokemonSpeciesTO {
        return PokemonSpeciesTO(
                pokedexNumber,
                name,
                getSpriteForGameAndPokemon(this, game, false),
                types,
                abilities,
                baseSpecies,
                evolutions,
                genderRate,
                isLegendary,
                generation
        )
    }
}

data class PokemonTypesTO(val primary: String, val secondary: String?)